compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -1144,10 +1147,11 @@ impl<'a, 'tcx, Prov: Provenance, Extra, Bytes: AllocBytes>
     pub fn write_scalar(&mut self, range: AllocRange, val: Scalar<Prov>) -> InterpResult<'tcx> {
         let range = self.range.subrange(range);
         debug!("write_scalar at {:?}{range:?}: {val:?}", self.alloc_id);
-        Ok(self
-            .alloc
-            .write_scalar(&self.tcx, range, val)
-            .map_err(|e| e.to_interp_error(self.alloc_id))?)
+        interp_ok(
+            self.alloc
+                .write_scalar(&self.tcx, range, val)
+                .map_err(|e| e.to_interp_error(self.alloc_id))?,
```

Comment:
`Ok(thing.map_err(...)?)` is odd. Why not just map the error with `.into` if that's even necessary at all

Reply:
The error has to be mapped because `alloc.write_scalar` returns a `AllocError`, and that has to be converted into an `InterpError`, which cannot be done with `Into` because the conversion requires knowing the `AllocId` (that's why those are different error types to begin with).

But yeah we can use `into()` here, I'll change that.

Reply:
I mean, after the `to_interp_error`... is `.into` necessary at all? Or at worst it would be `thing.map_err(|e| e.to_interp_error(...).into())` without `?` or `Ok`

Reply:
> `thing.map_err(|e| e.to_interp_error(...).into())`

How's that supposed to be possible? `thing` is a `Result` and we need an `InterpResult`. We can't `map_err` from one to the other.

Reply:
FWIW we have this pattern quite a few times so we should change them all if we keep the change to this one that I just made.

Reply:
Oh right. It was before your changes that this was a useless questionmark+OK op